tutorials/advanced-topics/basic-segment-tree.md

```diff
@@ -0,0 +1,179 @@
+---
+
+title: 'Segment Tree'
+
+description: 'Author: @Dhruvil-Lakhtaria'
+
+hide_table_of_contents: true
+
+---
+
+<TutorialAuthors  names="@Dhruvil-Lakhtaria"/>  
+
+### Introduction
+  Segment Tree is a type of datastructure used in very specific kind of problems. Problems that deal with 2 types of operations namely point `update()` and `query()`(Some kind of queries like Max, Min, GCD, etc.) for a given array.
+
+### Structure
+It is basically a binary tree with 3 types of nodes - 
+
+ 1. `Root Node` - denotes the whole array.
+ 2. `Leaf Node` - denotes a single element of the array.
+ 3. `Internal Node` - denotes a sub-interval of the array.
+ 
+### Representation
+We can implement the above using an array representation.
+Each Index denotes the answer to that interval with $$(2*i+1) th$$ index as its left child and $$(2*i+2)th$$ index as its right child.
+
+![Segment-Tree](https://miro.medium.com/max/1100/1*9V1l8_weUfmT4jHg21EE5w.png)
+
+
+
+### Segment Tree Implementation for Range Min Query.
+
+Lets look at [307. Range Sum Query - Mutable](https://leetcode.com/problems/range-sum-query-mutable/) .
+> You are given an array with n values. You have to answer two type of operations on it.
+>
+> 1. Get sum of values in range $$[L-R]$$.
+> 2. Update ith element to new value. 
+
+### Approach Using Segment Tree.
```

Comment:
1. Missing Query part.
2. for each section, add a code snippet to explain. at the end, show the full segment tree template (which you did below already)

Reply:
I dont think adding code snippet again here would help as the function name are clear enough and people can reaf there  itself.